Updated Error Section

tjlytle opened the pull request:
- Added new error 'type'.
- Updated type desicrptions a bit.
- Clarified role of client library in bridging old and new APIs.

tjlytle committed:
Updated error section:
- Added new error 'type'.
- Updated type desicrptions a bit.
- Clarified role of client library in bridging old and new APIs.

timcraft commented on SPECIFICATION.md:
```diff
@@ -108,10 +108,15 @@ Interactions with the API may result in an error, these may be transport errors,
- 	* Validation Errors (unexpected data from API).
+- Should have a type that classifies the error:
+    * Transport Errors (could not reach the API, did not receive a response from the API).
+    * Request Errors (invalid data, bad credentials, API rejected due to invalid request, 400 class errors).
```
Recommend changing "Request Errors" to "Client Errors" to match HTTP language.

tjlytle replied:
If we do that, "API errors" should be "Server errors" which works for me. 

Good catch as these names should influence the names in the client libraries.

timcraft replied:
Yep, ClientError for 4xx and ServerError for 5xx works well--matches the HTTP language, makes it clear where the fault lies, and easy to make more specific e.g. AuthenticationError is a ClientError.

tjlytle committed:
Clarified Errors
- Using Server / Client to match HTTP.
- Closes #4, having the library pass on the specific codes / messages.
